internals/cli/cmd_run.go

```diff
@@ -154,6 +157,9 @@ func runDaemon(rcmd *cmdRun, ch chan os.Signal, ready chan<- func()) error {
 	if err != nil {
 		return err
 	}
+	if rcmd.Dry {
```

Comment:
Shouldn't this go all the way to the sanityCheck below, right before Start?

Reply:
I've extensively discussed this with @flotter, and one of his (rightful) concerns where around system-wide side effects caused by e.g. listening on a port, since it's an "external" side effect (i.e. contributes to Pebble initialization by persisting some state externally, be it listening on a socket or calling `.Checkpoint()` on the state backend).

The idea is for the validation to be contained in the `.DryStart()` methods, and also `sanityCheck()` is a no-op for now anyway.